src/Grok.Net/Grok.cs

```diff
@@ -15,28 +15,30 @@ public class Grok
         private readonly Dictionary<string, string> _typeMaps;
         private Regex _compiledRegex;
         private List<string> _groupNames;
+        private readonly RegexOptions _regexOptions;
 
         private static readonly Regex _grokRegex = new Regex("%{(\\w+):(\\w+)(?::\\w+)?}", RegexOptions.Compiled);
         private static readonly Regex _grokRegexWithType = new Regex("%{(\\w+):(\\w+):(\\w+)?}", RegexOptions.Compiled);
         private static readonly Regex _grokWithoutName = new Regex("%{(\\w+)}", RegexOptions.Compiled);
         
-        public Grok(string grokPattern)
+        public Grok(string grokPattern, RegexOptions options = RegexOptions.Compiled | RegexOptions.ExplicitCapture)
         {
             _grokPattern = grokPattern;
             _patterns = new Dictionary<string, string>();
             _typeMaps = new Dictionary<string, string>();
+            _regexOptions = options;
             
             LoadPatterns();
         }
 
-        public Grok(string grokPattern, Stream customPatterns)
-            :this(grokPattern)
+        public Grok(string grokPattern, Stream customPatterns, RegexOptions options = RegexOptions.Compiled | RegexOptions.ExplicitCapture)
```

Comment:
This will add the possibility to not specify `RegexOptions.Compiled | RegexOptions.ExplicitCaptur`
```
   RegexOptions options = RegexOptions.Singleline;
   var grok = new Grok($"", options);
```

Reply:
yes, please. here it should be `RegexOptions regexOptions = RegexOptions.None`
but here instead of ` _compiledRegex = new Regex(pattern, _regexOptions);` 
we need
` _compiledRegex = new Regex(pattern, _regexOptions | RegexOptions.Compiled | RegexOptions.ExplicitCapture);`

Reply:
Just to confirm - you want the following `public` constructors?

```csharp
// RegexOptions RegexOptions.Compiled | RegexOptions.ExplicitCapture
public Grok(string grokPattern);

// RegexOptions RegexOptions.Compiled | RegexOptions.ExplicitCapture
public Grok(string grokPattern, Stream customPatterns);

// RegexOptions RegexOptions.Compiled | RegexOptions.ExplicitCapture
public Grok(string grokPattern, IDictionary<string,string> customPatterns);

// RegexOptions a concatination of passed in options AND RegexOptions.Compiled | RegexOptions.ExplicitCapture
public Grok(string grokPattern, IDictionary<string,string> customPatterns, RegexOptions options);
```

Will make the change tomorrow :)

Reply:
yes, exactly 👍